### PR TITLE
TRELLO-1173 Change start end param type (WIP)

### PR DESCRIPTION
**Pull request by eletallbetagouv**

### CODE REVIEW

File: app/utils/DateUtils.scala [hunk 1]
```diff
@@ -5,20 +5,30 @@ import java.time.LocalDate
...
   def parseDate(source: Option[String]): Option[LocalDate] =
     try source.map(s => LocalDate.parse(s, FORMATTER))
     catch {
       case _: DateTimeParseException => None
     }
 
+  def parseZonedDateTime(source: Option[String]): Option[ZonedDateTime] =
+    try source.map(s => ZonedDateTime.parse(s, ZONEDATETIME_FORMATTER))
+    catch {
+      case _: DateTimeParseException => None
```
Comment (ssedoudbgouv) [inline app/utils/DateUtils.scala:29]: Je me demande si c'est pas mieux de renvoyer un Try[ZoneDateTime], passer en None ça masque l'erreur du coup un utilisateur qui passe une date mal formatée ne verra rien côté API.

File: app/controllers/ReportListController.scala [hunk 2]
```diff
@@ -56,17 +58,25 @@ class ReportListController(
   def extractReports = SecuredAction(WithPermission(UserPermission.listReports)).async { implicit request =>
```
Comment (ssedoudbgouv) [inline app/controllers/ReportListController.scala:58]: There is a dedicated error object MalformedQueryParams (AppError used for this purpose) that will return 400 HTTP code.

I would have done something like this :

    for {
      reportFilter <- ReportFilter
        .fromQueryString(request.queryString, request.identity.userRole)
        .liftTo[Future] {
          logger.warn(s"Failed to parse ReportFilter query params")
          throw MalformedQueryParams
        }
      _ = logger.debug(s"Parsing zone query param")
      zone <- (new QueryStringMapper(request.queryString))
        .timeZone("zone")
        .liftTo[Future] {
          logger.warn(s"Failed to parse zone query param")
          MalformedQueryParams
        }
      _ = logger.debug(s"Requesting report for user ${request.identity.email}")
      file <- asyncFileRepository
        .create(AsyncFile.build(request.identity, kind = AsyncFileKind.Reports))
      _ = reportsExtractActor ! ReportsExtractActor.ExtractRequest(file.id, request.identity, reportFilter, zone)
    } yield Ok

File: app/controllers/ReportToExternalController.scala [hunk 3]
```diff
@@ -59,8 +59,8 @@ class ReportToExternalController(
     val qs = new QueryStringMapper(request.queryString)
     val filter = ReportFilter(
       siretSirenList = qs.string("siret").map(List(_)).getOrElse(List()),
-      start = qs.localDate("start"),
-      end = qs.localDate("end"),
+      start = qs.timeWithLocalDateRetrocompatStartOfDay("start"),
```
Comment (ssedoudbgouv) [inline app/controllers/ReportToExternalController.scala:62]: FYI I have shared  the external API documentations

File: app/tasks/report/ReportNotificationTask.scala [hunk 4]
```diff
@@ -45,13 +45,19 @@ class ReportNotificationTask(
 
   def runPeriodicNotificationTask(taskDate: LocalDate, period: Period): Future[Unit] = {
```
Comment (ssedoudbgouv) [inline app/tasks/report/ReportNotificationTask.scala:46]: What is the use for taskDate ? I think it may not be needed anymore

**Review by ssedoudbgouv — CHANGES_REQUESTED** [review]

File: app/tasks/report/ReportNotificationTask.scala (hunk 4 above)
Comment (eletallbetagouv) [reply app/tasks/report/ReportNotificationTask.scala]: Bien vu, c'était utilisé encore pour la date affichée en haut du mail. Il n'y a pas de raison de générer la date courante deux fois, je factorise

**Pull request merged**